### PR TITLE
[Internal] Query: Adds ability to request plain-text response for IndexMetrics

### PR DESCRIPTION
**Pull request by leminh98**
# Pull Request Template

## Description

This PR changes the default header sent to the backend when user requested IndexMetrics to the backend so that from now on the backend will send back plaintext response for IndexMetrics instead of base64 encoded. No change from end user, and not a breaking change

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by sboshra — DISMISSED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs [hunk 1]
```diff
@@ -183,7 +183,7 @@ class QueryResponse<T> : FeedResponse<T>
                 cosmosArray: cosmosElements,
                 serializerCore: serializerCore);
 
-            this.IndexUtilizationText = ResponseMessage.DecodeIndexMetrics(responseMessageHeaders, true);
+            this.IndexUtilizationText = ResponseMessage.DecodeIndexMetrics(responseMessageHeaders, false);
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs:186]: >false [](http://example.com/codeflow?start=99&length=5)

Named parameter would be nice.

**Review by adityasa — APPROVED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 2]
```diff
@@ -260,7 +260,7 @@ internal override void PopulateRequestOptions(RequestMessage request)
 
             if (this.PopulateIndexMetrics.HasValue)
             {
-                request.Headers.CosmosMessageHeaders.Add(HttpConstants.HttpHeaders.PopulateIndexMetrics, this.PopulateIndexMetrics.ToString());
+                request.Headers.CosmosMessageHeaders.Add(HttpConstants.HttpHeaders.PopulateIndexMetricsText, this.PopulateIndexMetrics.ToString());
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:263]: #if IndexAdvisorV2
request.Headers.CosmosMessageHeaders.Add("<<New Header>>", this.PopulateIndexMetrics.ToString());
#else
request.Headers.CosmosMessageHeaders.Add(HttpConstants.HttpHeaders.PopulateIndexMetrics, this.PopulateIndexMetrics.ToString());
#endif

**Pull request closed without merging**